domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/searchaliasfield/SearchAliasFieldTypeValidator.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.searchaliasfield;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.SearchAliasFieldRepository;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.SearchAliasFieldEntity;
+
+@Component
+public class SearchAliasFieldTypeValidator implements SearchAliasFieldValidator {
+
+    private final SearchAliasFieldRepository repository;
+
+    @Autowired
+    public SearchAliasFieldTypeValidator(SearchAliasFieldRepository repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public ValidationResult validate(SearchAliasFieldEntity searchAliasField) {
+
+        ValidationResult validationResult = new ValidationResult();
+
+        if (searchAliasField.getFieldType() == null) {
+            validationResult.addError(new ValidationError(String.format("Invalid case field '%s' for search alias ID '%s' and case type '%s'. Case field should"
+                                                                            + " point to a concrete field with full object notation in case of a complex type.",
+                                                                        searchAliasField.getCaseFieldPath(),
+                                                                        searchAliasField.getReference(),
+                                                                        searchAliasField.getCaseType().getReference()),
+                                                          searchAliasField));
+        } else {
+            repository.findByReference(searchAliasField.getReference())
+                .forEach(field -> {
```

Comment:
can you rename field to alias? it's confusing otherwise

Reply:
as a bit more readable alternative we could use
boolean sameType = repository.findByReference(searchAliasField.getReference()).stream().allMatch(.....)
if (!sameType) throw...

Reply:
renamed field 👍 

Cant use allMatch as I need access to the existing alias field so the error message can describe what is the mismatch. I am now using findFirst though to make is more readable. ta